src/casing.rs

```diff
@@ -0,0 +1,61 @@
+use onig::Regex;
+
+/// Normalize the casing of a string to be `snake_case`.
+///
+/// This function produces strings that are transformed consistently from a
+/// variety of different input casing. The rule-set for word boundaries are
+/// derived from the withoutboats/heck crate. Underscores are considered word
+/// boundaries in addition to the standard pattern e.g. `\b`. `camelCasing` is
+/// detected by a lowercase followed by an uppercase. Numbers can take on either
+/// case depending on the preceeding symbol.
+///
+/// See: https://github.com/withoutboats/heck/blob/master/src/lib.rs#L7-L17
```

Comment:
Maybe better to use a fixed-point-in-time link: https://github.com/withoutboats/heck/blob/093d56fbf001e1506e56dbfa38631d99b1066df1/src/lib.rs#L7-L17